File: Assets/Scripts/Objects/Pollen.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Pollen : Network.NetworkObject
{
    /// <summary>
    /// Drop the pollen (if on base, add it)
    /// </summary>
    /*public override void Drop()
    {
        if (_owner == null) return;//Todo : Fix owner null
        Vector2Int indexes = HexaGrid.WordPositionToHexIndexes(_owner.transform.position);
        Base baseOn = GameManager.Instance.HexaGrid.GetPropertyOfHexIndex(indexes);
        if(baseOn != null)
        {
            baseOn.Upgrade(1);
            OnDestroyNeeded();
        }
        base.Drop();
    }*/
}

File: Server/src/objects/pollen.ts
import HexaGrid from "../hexagrid";
import NetworkObject from "./networkObject";

/**
 * A pollen is an object that can be dropped on a base to upgrade it
 */
export default class Pollen extends NetworkObject{
    /**
     * Drop the pollen. If on a base, upgrade it by one level (one hexagon).
     */
    public override drop(): void {
        let indexes = HexaGrid.wordPositionToHexIndexes(this._owner!.currentSimulationState.position);
        let baseOn = this._networkManager.hexaGrid.getPropertyOfHexIndex(indexes);
        if(baseOn != null){
            baseOn.upgrade(1);
            super.destroy();
        }
        super.drop();
    }
}